spec/lib/task-spec.js

```diff
@@ -508,6 +502,64 @@ describe("Task", function () {
 
     });
 
+    describe("timer", function() {
+        it("should not timeout when value is 0", function() {
+            var definition = _.cloneDeep(noopDefinition);
+            definition.options.timeout = 0;
+            var task = Task.create(definition, {}, {});
+            task.renderAll = sinon.stub().resolves();
+            sinon.spy(task, 'cancel');
+
+            return task.run().then(function() {
+                expect(task.state).to.equal('succeeded');
+                expect(task.error).to.equal(null);
+            });
+        });
+
+        it("should not timeout when value is -1", function() {
+            var definition = _.cloneDeep(noopDefinition);
+            definition.options.timeout = -1;
+            var task = Task.create(definition, {}, {});
+            task.renderAll = sinon.stub().resolves();
+            sinon.spy(task, 'cancel');
+
+            return task.run().then(function() {
+                expect(task.state).to.equal('succeeded');
+                expect(task.error).to.equal(null);
+            });
+        });
+
+        it("should timeout with options.timeout", function() {
```

Comment:
I think @benbp is just making sure when the passed in timeout equals 1ms that the state equals `"timeout"` with the correct error message indicating the timed out value.  It's not a functional test that requires real delays?